lib/puppet/type/cisco_aaa_authorization_login_cfg_svc.rb

```diff
@@ -0,0 +1,113 @@
+# Manages configuration for Aaa Authorization Login Config Service.
+#
+# December 2015
+#
+# Copyright (c) 2013-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_aaa_authorization_login_cfg_svc) do
+  @doc = "Manages configuration for Authorization Login Config Service.
+
+~~~puppet
+  cisco_aaa_authorization_login_cfg_svc {\"[console|default]\":
+    ..attributes..
+  }
+~~~
+
+  'console' and 'default' are the only services configurable.
+
+  Example:
+~~~puppet
+    cisco_aaa_authorization_login_cfg_svc {\"console\":
+      ensure    => present,
+      groups    => ['group1', 'group2'],
+      method    => \"local\",
+    }
+~~~"
+
+  ensurable
+
+  ###################
+  # Resource Naming #
+  ###################
+
+  # Parse out the title to fill in the attributes in these patterns.
+  # These attributes can be overwritten later.
+  def self.title_patterns
+    identity = ->(x) { x }
+    patterns = []
+
+    patterns << [
+      /^(\S+)$/,
+      [
+        [:name, identity]
+      ],
+    ]
+    patterns
+  end
+
+  newparam(:name, namevar: true) do
+    desc 'Name of the config login service. Valid values are "console" or "default".'
+
+    validate do |value|
+      fail unless value == 'default' || value == 'console'
+    end
+  end
+
+  ##############
+  # Attributes #
+  ##############
+
+  newproperty(:groups, array_matching: :all) do
```

Comment:
For properties that use a single dimensional arrays we have started support both of the following with the hope that it's somewhat more user friendly when creating the manifest.

1) ['string1', 'string2']
2) "string1 string2"

Here is an example of the type code needed to support both.

https://github.com/cisco/cisco-network-puppet-module/blob/develop/lib/puppet/type/cisco_bgp_af.rb#L573

Only minor changes needed to split and process.